# Pre-Eclampsia and Hypertension

In [11]:
from opyn.generic.pandasloader import PandasLoader
from opyn.stats.observationalstudies import TwoByTwo

## Summary

The goal of this exercise was to measure and test the association between a woman having suffered from pre-eclampsia or eclampsia during their first pregnancy and them developing hyptertension in later life.
Data from a cohort study were obtained from a published source and analysed using standard epidemiological methods.
We found there was strong evidence of a positive association between the two events.

## Introduction

Researchers followed 542 women who suffered from pre-eclampsia and eclampsia during their first pregnany and 277 women who did not over time, and counted those who developed hypertension in later life and those who did not.
The issue of interest in the reserch was whether there was evidence of an association between the pre-clampsia and eclampsia, and hypertension.
The data for the analysis were taken from Wilson, B.J., Watson, M.S., Prescott, G.J. *et al* (*BMJ* 2003;326:845).
(See description for full reference.)

## Initialisation

We use `PandasLoader` to get the data and print the description.

In [12]:
pdloader = PandasLoader()

In [13]:
f = 'preeclampsia'
data = PandasLoader().get(f)
pdloader.get_description(f)

title='Pre-eclampsia and Eclampsia and Hypertension in later life'

description='Results of a cohort study looking at the association between
             pre-eclampsia and eclampsia during a woman's first pregnancy, and
             the development of hypertension in later life'

references='Wilson, B. J., Watson, M. S., Prescott, G. J., Sunderland,
            S., Campbell, D. M., Hannaford, P., & Smith, W. C. S. (2003).
            Hypertensive diseases of pregnancy and risk of hypertension and
            stroke in later life: results from cohort study. BMJ, 326(7394),
            845. https://doi.org/10.1136/bmj.326.7394.845'

has_bibtex=True@Wilson2003

fields:
    count, int: Number of observation.
    exposure, str: Did the the patient have pre-eclampsia.
                   One of either ['pre-eclampsia', 'no pre-eclampsia'].
    outcome, str: Did the the patient have hypertesion in later life
                  One of either ['hypertension', 'no hypertension'].



We prepare the dataframe by replacing the nominal labels with integers, so the **exposure, disease** is replaced by `1` and **no exposure, no disease** by `2`.
This ensures that the `TwoByTwo` object initialises as expected.

In [14]:
new_labels = {'pre-eclampsia': 1,
              'no pre-eclampsia': 2,
              'hypertension': 1,
              'no hypertension': 2}
data.replace(new_labels, inplace=True)
tbl = TwoByTwo.from_dataframe(data)

## Results

The results of the study as a two-by-two contingency tables are given below in `Table 1`.

In [21]:
print("Table 1.")
print(tbl.show(row_totals=True))

Table 1.
Outcome      Disease  No Disease  Total
Exposure                               
Exposed          327         215    542
Not Exposed       76         201    277


### Strength of assocation

The relative risk is **2.20**, with **95%** confidence interval **(1.79, 2.69)**.

In [22]:
print(tbl.relative_risk())

              estimate       ese       lcb      ucb
RelativeRisk  2.198946  0.103735  1.794385  2.69472


The odds ratio is **4.02**, with **95%** confidence interval **(2.94, 5.51)**.

In [23]:
print(tbl.odds_ratio())

           estimate       ese       lcb       ucb
OddsRatio   4.02246  0.160755  2.935327  5.512225


The relative risk and the odds ratio are both greater than **1**, with both confidence intervals located well above **1**.
This suggests there is a postivie association between a patient having pre-eclampsia or eclampsia during the first pregnancy and developing hypertension in later life.

### Test of No Association

The frequencies expected under the null hypothesis of no association are given below in `Table 2`.

In [24]:
print("Table 2.")
print(tbl.expected_freq(row_totals=True))

Table 2.
Outcome         Disease  No Disease  Total
Exposure                                  
Exposed      266.698413  275.301587  542.0
Not Exposed  136.301587  140.698413  277.0


The number of observed patients who developed hypertension whilst having had pre-eclampsia and eclampsia during their first pregnancy (**327**) is much greater than the expected frequency (**266.70**).

We can take the approximation to be adequate, given all expected frequencies are at least **5**.

The **Chi-squared** contributions for each observation are shown in `Table 3`.

In [25]:
print("Table 3.")
print(tbl.chi2_contribs())

Table 3.
Outcome        Disease  No Disease
Exposure                          
Exposed      13.634432   13.208356
Not Exposed  26.678203   25.844509


The test statistic is **79.37** on **1** degree of freedom, with **p**-value of less than **0.000**.

In [26]:
print(tbl.chi2_test())

               chisq          pval  df
ChiSqTest  79.365501  5.161965e-19   1


Given **p** < 0.001, there is strong evidence against the null hypothesis of no association.
There is strong evidence of an association between a woman suffering from pre-eclampsia and eclampsia during their first pregnancy and them developing hypertension in later life.

## Discussion

The study found there to be strong evidence (**p < 0.001**) for a positive association between a woman having had pre-eclampsia or eclampsia during their first pregnancy and them developing hypertension in later life (**OR = 4.02**).

Women who had from pre-eclampsia or eclampsia during their first pregancy were found to be **119%** more likely to develop hypertension than those women who did not.